bredis/AsyncConnection.hpp

```diff
@@ -51,24 +53,75 @@ template <typename S> class AsyncConnection {
     std::mutex tx_queue_mutex_;
     std::mutex rx_queue_mutex_;
 
+    // I think that queuing should be the responsibility
+    // of a higher level interface and not this class.
+    // Instead, this class should just allow for one
+    // active asynchronous Redis read and one active
+    // asynchronous Redis write, with undefined behavior
+    // if two concurrent reads or two concurrent writes
+    // happen.
+    //
+    // Rationale: TCP/IP provides application-level flow
+    // control to match the rate of the receiver and the
+    // sender. In this implementation, the reader attempts
+    // to read as quickly as possible no matter how fast
+    // the consumer (in this case the command_callback_t)
+    // can process the data. This not only circumvents
+    // TCP/IP's natural application level flow control,
+    // but allows the size of the respective queues to
+    // grow without bounds. Furthermore, the caller is
+    // given no opportunity to provide "back pressure",
+    // nor is the caller able to detect the unbounded
+    // growth and impose its own policies on it.
+    // 
```

Comment:
> You shouldn't wait for the corresponding read to complete before sending the next ping.

https://github.com/basiliscos/cpp-bredis/blob/try-avoid-queuing/t/11-multi-ping.cpp#L62

In the new (sample) implementation i don't, i.e. next `ping` is pushed as soon as previous has been written. 

On the other hand, I think the new `async_write` can be generalised to accept **multiple** commands with their arguments. I.e. like 

```
push_write(command_with_args, callback);
push_write(vector<command_with_args>, callback);
```

it will do the exactly same, as I did in the `v0.01`, i.e. created one big TX-buffer and pushed it to system.

Will try to experiment with disabling Nagle algorithm, though.